wallet/createtx.go

```diff
@@ -41,15 +33,23 @@ func makeInputSource(eligible []wtxmgr.Credit) txauthor.InputSource {
 		[]btcutil.Amount, [][]byte, error) {
 
 		for currentTotal < target && len(eligible) != 0 {
-			nextCredit := &eligible[0]
```

Comment:
Not sure why before a pointer was obtained here, just to then get the addr again to call into `NewTxIn`...

Reply:
Yeah. Looking closer at this function also makes me think something's pretty weird in the way it is implemented.
If we can call the input source multiple times but keep the slices and total around, wouldn't that mean that we could never select _fewer_ inputs in a subsequent call that we've selected in a previous call?
Not sure if the underlying coin selection algorithm actually operates under the same assumption...